src/legacy/core_plugins/vis_default_editor/public/components/agg_params.tsx

```diff
@@ -76,11 +78,19 @@ function DefaultEditorAggParams({
   setTouched,
   setValidity,
 }: DefaultEditorAggParamsProps) {
-  const groupedAggTypeOptions = useMemo(() => getAggTypeOptions(agg, indexPattern, groupName), [
-    agg,
-    indexPattern,
-    groupName,
-  ]);
+  const {
+    services: {
+      dataShim: {
+        search: {
+          aggs: { types: aggTypes, aggTypeFieldFilters },
```

Comment:
What do you think about getting rid of `aggTypeFieldFilters` in the data search services?
The only thing it is responsible for is adding a filter in `rollup` and using it in the default editor.
This is pretty the same thing as we've already got rid of `editorConfig` registry in #56501
This is maybe a question to @lukeelmers

Reply:
I agree, this would simplify the setup as more stuff becomes static. We still need a stateful plugin for `types` to keep it extensible, but it takes away complexity. Also cc @ppisljar 

Have there been thoughts about this already?

Reply:
I'm currently working out the long term interface for agg types in #57064, which I'm hoping should be ready to merge by next week.

So far the items we will for sure be keeping in the lifecycle contract are `types` (registry of agg types) and `createAggConfigs` (a function that returns a `new AggConfigs()`, which will make the `AggConfigs` class itself internal to the data plugin).

Everything else is temporarily under a `__LEGACY` namespace while work out whether they are required or if they can be relocated/removed. For example, `aggTypeFieldFilters` is currently `data.search.aggs.__LEGACY.aggTypeFieldFilters`, but I agree @sulemanof, it may not be necessary to keep at all.

#57064 will be the best place to follow along, as many of these details should be sorted out in the coming days. Would welcome any input you have on that @sulemanof @flash1293 ... I'll probably add you as reviewers once I get it wrapped up 😄

Reply:
As it's not urgent I will postpone work on this PR till @lukeelmer's is merged. @kertal I will ping you once it becomes ready again :)

Reply:
Just as an update here: #58462 and #58805 have both been merged this week, and I'm actively working now on the PR to cutover agg types to the new platform.

Since doing the cutover is the highest priority in terms of unblocking teams, we decided to postpone sorting out `aggTypeFieldFilters` and some of the other items under `__LEGACY` as it should be possible to address them after the service is in the new platform, since they are no longer relying on legacy world functionality.

As we clean up those APIs we will upstream affected downstream applications accordingly.

Reply:
Makes sense, as there are some parts that definitely will stay stateful anyway it's a very small change to make later on.